akka-stream/src/main/scala/akka/stream/impl/StreamLayout.scala

```diff
@@ -98,20 +116,28 @@ import scala.util.control.NonFatal
   import VirtualProcessor._
 
   override def toString: String = s"VirtualProcessor(${this.hashCode()})"
+  if (VirtualProcessor.Debug) println(s"created: $this")
 
   override def subscribe(s: Subscriber[_ >: T]): Unit = {
-    @tailrec def rec(sub: Subscriber[Any]): Unit =
+    @tailrec def rec(sub: Subscriber[Any]): Unit = {
       get() match {
-        case null ⇒ if (!compareAndSet(null, s)) rec(sub)
+        case null ⇒
+          if (VirtualProcessor.Debug) println(s"VirtualPublisher#$hashCode(null).subscribe.rec($s) -> sub")
```

Comment:
what does `VirtualPublisher#$hashCode` mean in these prints?

no `$` for `sub`?

Reply:
Just wanted to identify different `VirtualPublisher` instances, so that's why I included the hashcode, when I was trying to figure it out. I think default java Object does a "Type#hashcode" when it `toString`s so mimicked that. The interesting part about sub is already in the string as `$s` is the actual subscription

Not 100% sure we need to leave the debugging in here at all.

Reply:
ah, now I see the $h...